File: BasicSortingAlgorithms/MazorityElement/boyer-moore-algo.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int mazorityEle = nums[0];
        int appearCount = 1;
        for(int i=1;i<n;i++){
            if(nums[i]!=mazorityEle){
                appearCount--;
                if(appearCount == 0){
                    mazorityEle = nums[i];
                    appearCount++;
                }
            }
            else appearCount++;
        }
        return mazorityEle;
    }
};

File: BasicSortingAlgorithms/MazorityElement/boyer-moore-algo.py
from typing import List

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        mazorityEle = nums[0]
        appearCount = 1
        for i in range(n):
            if nums[i]!=mazorityEle:
                appearCount-=1
                if appearCount == 0:
                    mazorityEle = nums[i]
                    appearCount+=1
            else: appearCount+=1
        return mazorityEle